raco/backends/myria/myria.py

```diff
@@ -790,6 +828,53 @@ def compileme(self, inputid):
         }
 
 
+class MyriaConsumer(algebra.ZeroaryOperator, MyriaOperator):
```

Comment:
This new class is again for breaking the infinite loop of optimization. The query plan being constructed has loops such as: `IDBController` -> (sending EOIs) `Consumer` -> `EOSController` -> (sending EOSs) `Consumer` -> 'IDBController`, so I would need to break the loop to stop the recursive optimization. I ended up with making these consumers as `ZeroaryOperator` to set the optimization boundary. I agree that it's not a neat refactoring but given the current design, I'm not sure if a better solution exists. Maybe we can rename it to be more clear. What do you think?